### ISSUE TITLE
Credit card numbers are improperly formatted

### ISSUE THREAD

**Issue by warsaw** [bitwarden/clients#11977]
### Steps To Reproduce

1. Go to 'Card'
2. Click on a Visa or Mastercard
3. View the Number

### Expected Result

I'd expect to see a number formatted like `1234 5678 9012 3456`, i.e. four groups of 4 numbers.

### Actual Result

I see a number like `1234 567 8 90 12 3 456`.   This happens even if edit the card, delete the spaces and add them back in the right place.

### Screenshots or Videos

Unavailable due to sensitive nature of actual numbers.

### Additional Context

This doesn't happen for all cards.  Some format just fine.  I've only seen this for Visa and Mastercard numbers, but some Visa and Mastercard fields are fine.  There's no commonality that I can tell.  Only the macOS desktop version seems to be affected.  iOS and iPadOS versions seem fine.

### Operating System

macOS

### Operating System Version

14.7.1

### Installation method

Mac App Store

### Build Version

2024.10.2 (32149)

### Issue Tracking Info

- [X] I understand that work is tracked outside of GitHub. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by rmcdowell-bitwarden** [comment]
Hi there,

Thank you for your report! 

I was able to reproduce this issue, and I have flagged this to our engineering team.

If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.

Thanks once again!

**Comment by djsmith85** [comment]
Hi @warsaw and thank you for your report.

While fixing https://github.com/bitwarden/clients/issues/12505 for the new UI's I did also check the desktop client and the spacing appears correct.

The spacing is defined within [credit-card-number.pipe.ts](https://github.com/bitwarden/clients/blob/main/libs/angular/src/pipes/credit-card-number.pipe.ts) and supports a few common brands. 

- Is your credit-card maybe from an unsupported brand? 
- Or does the number not fit the specified pattern?

Kind regards,
Daniel

**Comment by warsaw** [comment]
Hi @djsmith85 

No, there's nothing odd or nonstandard about the cards I'm seeing the problem with.  A perfect example is an Apple Card, brand Mastercard.  On iOS and iPadOS Bitwarden, when you toggle reveal, the number shows up perfectly well as four four-digit spans separated by spaces.  On macOS with Bitwarden client `2024.12.1 (34750)` updated from the App Store, it shows up as `1234 567 8 90 12 3 456`.  Verified on both multiple macOS machines (both using Bitwarden from the App Store).  I even tried editing the entry, completely deleting the number and retyping it as four four-digit spans, but again, it still shows up as above.  I even tried sending the entire entry to the trash and entering an entirely new item information, and that *still* shows the card number incorrectly.  Of course, syncing the vault on macOS and then on iOS shows it correctly on iOS.

**Comment by warsaw** [comment]
Also, looking at the code you referenced, I don't see the block layout I'm seeing described in that record at all.

**Comment by warsaw** [comment]
One other interesting data point.  If you hit the copy icon on a badly formatted number in the macOS client, then paste the number, you *do* seem to get a correctly formatted 4x4 block of numbers.  That in its own right can be problematic since not every website accepts spaces between blocks, but it's still interesting that even though the number is incorrectly formatted in the UI, it still at least pastes correctly.

**Comment by warsaw** [comment]
I just noticed the [release announcement](https://github.com/bitwarden/ios/releases) for 2025.9.0 (2578) on iOS:

> Credit card numbers now display in groups of 4 digits for easier readability.

I just updated my iPadOS client and confirm that I'm running this version.  And indeed, my credit card numbers are being displayed correctly on the iPadOS app (well, except for Amex which doesn't have 16 digits).  The same is *not* true for the macOS desktop client 2025.9.0 (47196), which still shows the credit card number formatting bug reported here.

**Comment by pamperer562580892423** [comment]
@warsaw Another user here. - Regarding the desktop app: did you ever check if the numbers were entered with spaces? I had this error also on the desktop app once, and it turned out to be space characters I wasn't aware of.

**Comment by warsaw** [comment]
Yep, when I first reported the bug I tried re-entering the number with the correct spacing multiple times, but it never fixed the display.  I could never figure out *why* the numbers were displayed wrong and eventually gave up.  I suppose I should try again now.

**Comment by warsaw** [comment]
Here's another funny thing about the desktop app: if you edit the card (with the pencil) and reveal the number in the item details edit panel, it actually shows up correctly!  But then even if you save it, once edited, it still shows up weirdly formatted.  I don't get it! 😄

**Comment by pamperer562580892423** [comment]
> Yep, when I first reported the bug I tried re-entering the number with the correct spacing multiple times

Haha, okay! But did you try it without entering any spaces? (the correct spacing is zero space characters)

**Comment by warsaw** [comment]
> But did you try it without entering any spaces? (the correct spacing is zero space characters)

No I didn't, but I just did and it worked!  So maybe it's a data-entry bug?  The edit form certainly shouldn't care about extraneous spaces.

**Comment by warsaw** [comment]
Okay, so the workaround is to edit the item, and just delete the spaces. Then when you save it, the numbers show up in groups of 4.
